src/Controller.cpp

```diff
@@ -176,7 +177,8 @@ namespace geopm
                      nullptr,
                      environment().endpoint(),
                      environment().do_endpoint(),
-                     ApplicationSampler::default_shmkey())
+                     ApplicationSampler::default_shmkey(),
+                     InitControl::make_unique(environment().init_control()))
```

Comment:
Switch this around so:
1. InitControl is always created.  Remove the input file string from the factory method.
2. Add a method to the Environment singleton for `do_init_control`.  If it is_set, then call parse_input with the input filename string.
3. In all cases, call write_controls().  It will be a no-op when do_init_control was false.